lisa/sut_orchestrator/baremetal/cluster/idrac.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import time
+import xml.etree.ElementTree as ETree
+from typing import Any, Type
+
+import redfish  # type: ignore
+
+from lisa import schema
+from lisa.environment import Environment
+from lisa.util import LisaException
+from lisa.util.logger import get_logger
+from lisa.util.perf_timer import create_timer
+
+from ..schema import ClusterSchema, IdracSchema
+from .cluster import Cluster
+
+
+class Idrac(Cluster):
+    def __init__(self, runbook: ClusterSchema) -> None:
+        super().__init__(runbook)
+        self.idrac_runbook: IdracSchema = self.runbook
+        self._log = get_logger("idrac", self.__class__.__name__)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "idrac"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return IdracSchema
+
+    def deploy(self, environment: Environment) -> Any:
+        self.redfish_instance = redfish.redfish_client(
+            base_url="https://" + self.idrac_runbook.address,
+            username=self.idrac_runbook.username,
+            password=self.idrac_runbook.password,
+        )
+        self._login()
+        self._change_boot_order_once("VCD-DVD")
+        for client in self.idrac_runbook.client:
```

Comment:
Usually, iDRAC is dedicated to a server which means there would be only one client per idrac_runbook. So may be we can enfore only one entry for `clients` section in the schema and remove the loop here.

Reply:
updated